File: algorithm/sort/selection/selection.go
// go run selection.go
package main

import "fmt"

func main()  {
	arr := selectionSort([]int{1, 3, 4, 2, 5})
	fmt.Printf("selection sort result %v", arr)
}

func selectionSort(arr []int) []int {
	length := len(arr)
	var minIndex int

	for i := range arr {
		minIndex = i

		for j := i + 1; j < length; j++ {
			if arr[minIndex] > arr[j] {
				minIndex = j
			}
		}

		arr[i], arr[minIndex] = arr[minIndex], arr[i]
	}
	return arr
}

File: algorithm/sort/selection/selection.js
// node selection.js
/**
 * selection sort
 * @param {Array} arr
 * @return {Array}
 */
function selectionSort(arr) {
    var len = arr.length;
    var minIndex, temp;

    for (var i = 0; i < len; i++) {
        minIndex = i;
        for (var j = i + 1; j < len; j++) {
            if (arr[minIndex] > arr[j]) {
                minIndex = j;
            }
        }
        temp = arr[i];
        arr[i] = arr[minIndex];
        arr[minIndex] = temp;
    }

    return arr;
}

console.log(selectionSort([1, 5, 3, 5, 3]))